### PR TITLE
chore(connector)_: Polish connector service

### PR DESCRIPTION
**Pull request by MishkaRogachev**
Changes: 
- [x] Minor code improvements
- [x] Add a signal for `wallet_switchEthereumChain`
- [x] Add a test proving that dApp can be reconnected several times

### CODE REVIEW

**Review by kounkou — APPROVED** [review]

**Review by jrainville — APPROVED** [review]

File: services/connector/api.go [hunk 1]
```diff
@@ -109,6 +109,7 @@ func (api *API) CallRPC(inputJSON string) (interface{}, error) {
 }
 
 func (api *API) RecallDAppPermission(origin string) error {
+	// TODO: close the websocket connection
```
Comment (alexjba) [inline services/connector/api.go:112]: What if there's another app connected to the browser connect?

Related questions (for me to better understand 😄):
1. How can we manage the connection and close it?
2. Shouldn't the connection be always active? Closing it won't stop any further data transmission between plugin and Status?

**Review by alexjba — APPROVED** [review]
👍 Thank you!

File: services/connector/api.go (hunk 1 above)
Comment (MishkaRogachev) [reply services/connector/api.go:112]: 1. That is the main issue currently. I know of two ways to fix this:
- Make changes to the go-ethereum code and add the ability to control connections using signals or something else. That's a very bad idea
- Add a new separate websocket server directly to the connector service. I think this is the right way to do it, at the same time we can remove nesting and escaping in json. 

2. Websocket server works in one-to-many way, allowing several clients to access in one time. To close the connection you need to select a connection with origin matching to dApp's url.

Comment (alexjba) [reply services/connector/api.go:112]: > Websocket server works in one-to-many way, allowing several clients to access in one time. To close the connection you need to select a connection with origin matching to dApp's url.

Does this mean the plugin will span a new connection for each dApp? I was assuming each plugin instance will establish a single connection for multiple dapps, acting like a proxy

Comment (MishkaRogachev) [reply services/connector/api.go:112]: Yes, for each DApp there is a websocket handler. Here is the pictures for the current approach and what i advise to do
Screenshot 2024-08-15 at 17 50 20

**Pull request merged**